coderd/mcp_http.go

```diff
@@ -9,10 +9,11 @@ import (
 	"github.com/coder/coder/v2/coderd/httpmw"
 	"github.com/coder/coder/v2/coderd/mcp"
 	"github.com/coder/coder/v2/codersdk"
+	"github.com/coder/coder/v2/codersdk/toolsdk"
 )
 
 // mcpHTTPHandler creates the MCP HTTP transport handler
-func (api *API) mcpHTTPHandler() http.Handler {
+func (api *API) mcpHTTPHandler(tools []toolsdk.GenericTool) http.Handler {
```

Comment:
I think we could simplify this a bit by making it a boolean called `isChatGPT bool`, then calling the appropriate Register function (either `.RegisterTools()` or `.RegisterChatGPTTools()`).

This also makes the handler less complex.